### PR TITLE
Reorganization of project files

### PR DESCRIPTION
**Pull request by TSRBerry**
As I started working on my next contribution I noticed that most of my commits aren't really related to the feature I want to implement, so I decided to PR them first and use that as a starting point. This also has the benefit that my next PR won't be too big.

Things I've done in this PR:

- Added MiniScript as a nuget package (depends on https://github.com/JoeStrout/miniscript/pull/62)
  -  This should also allow dependabot to create PRs for new MiniScript releases
- Added first bit of i18n localization
- Made loading assets cross-platform compatible as shown here: [SV-Wiki](https://stardewvalleywiki.com/Modding:Modder_Guide/Get_Started#How_do_I_make_my_mod_work_crossplatform.3F)
- Reorganized project files to make it easier to navigate
- Merged `SalableBot` into `BotObject`
- Added a debug key to solve all tasks instantly (this is only present in debug builds)
- ~~Created `ModDataExtensions` (10ea3412b967865589fee11efd8187346de833b8) - I'm still unhappy with them and might remove them again.~~ Removed.
- Added `ModData` class to manage save data (for now it's only used during conversion).
- Added customized `FarmerBot` to make things easier when workarounds are needed.
- Only destroy bot if it contains more items than just the initial tools

I'll mark this PR as ready for review as soon as the MiniScript PR has been dealt with and when I'm happy with the state of my PR.

I hope that I was able to conform to your code style for most of my commits.

### CODE REVIEW

**Comment by TSRBerry** [comment]
There are two things that don't work right now:

- You can't drop items into empty slots in a bot. (I'm not sure if this also happens in the current version)
- The scythe is not able to cut grass for some reason, but all other tools can.

What is left to do:

- [ ] Try to fix the bugs.
- [ ] Add a new intrinsic to change the name of the bot.
- [ ] If the bot contains items (not counting initial tools) either drop them or stop the bot from getting picked up.

**Comment by TSRBerry** [comment]
This PR got a lot bigger than I wanted, so I marked it as ready for review now, as most things are already done and only some small changes need to be made.

**Comment by TSRBerry** [comment]
Turns out there already was an `assignOverride` key present for `name`. I just changed the affected property to `BotName` which is internal to the mod. 
This way we can keep the item name the same and not deal with finding out which variable we would need to use. (Since either `DisplayName` or `Name` gets used by the game depending on the language.)

**Comment by JoeStrout** [comment]
Good fix.  And good catch on the wiki page, too (now corrected)!

**Comment by TSRBerry** [comment]
I have no idea why the scythe works now, but I just tested it again and didn't encounter any problems.

But that also means: This PR is finally completely ready!

File: Farmtronics/Bot/BotObject.cs [hunk 1]
```diff
@@ -213,23 +213,41 @@ class BotObject : StardewValley.Object {
...
 				// If we can get a terrain feature, then have it do the "use" action,
 				// by temporarily setting the bot farmer to be the Game1 player.
+				ModEntry.instance.Monitor.Log("Harvesting: TerrainFeature");
+				
 				var origPlayer = Game1.player;
 				Game1.player = farmer;
-				bool result = feature.performUseAction(tileLocation, loc);
+				bool result = loc.terrainFeatures[tileLocation].performUseAction(tileLocation, loc);
 				Game1.player = origPlayer;
 				return result;
```
Comment (TSRBerry) [inline Farmtronics/Bot/BotObject.cs:232]: What would be an example of a case that would enter this if-block?

Comment (JoeStrout) [reply Farmtronics/Bot/BotObject.cs]: Hmm, good question.  I should have left better comments!  But I think this is for taking items out of machines (which is mentioned in the 1.2 release notes as something that bot.harvest can do).

**Comment by JoeStrout** [comment]
OK, I confess, I'm disinclined to review 92 new/changed files.  :)  I've read through your comments and it all sounds reasonable, and I believe you do good work, so in it goes!  If anything breaks we can fix it later.  Thank you for all your hard work!

**Pull request merged**